src/map/packet_system.cpp

```diff
@@ -2554,7 +2554,14 @@ void SmallPacket0x04D(map_session_data_t* const PSession, CCharEntity* const PCh
 
             if (PChar->UContainer->IsSlotEmpty(slotID))
             {
-                int32 ret = sql->Query("SELECT charid, accid FROM chars WHERE charname = '%s' LIMIT 1;", str(data[0x10]));
+                std::string charName = str(data[0x10]);
+                if (!std::regex_match(charName, std::regex("^([A-Z]{1}[a-z]{2,14})$")))
```

Comment:
Remind me, this isn't one of the places where M.H.Whatever.Whatever can send you things?

Reply:
Delivery box/sendbox AH records can possibly have non alphanumeric characters in the name (in the AH's case, because everyone tends to name their AH bot something invalid intentional so that players can't take its name)

Reply:
retail has mailed things from `M.H.M.U.` (mog house management union) on occasion

Reply:
This isn't relevant here, this is packets recieved from the client about where to send it to, they can only send to real players, aka `^([A-Z]{1}[a-z]{2,14})$`

Reply:
cools then